### PR TITLE
Introduce `bypass_fsm_freeze` and remove the use of `updated_fields`

### PR DESCRIPTION
**Pull request by ming-tung**
issue https://github.com/ming-tung/django-fsm-freeze/issues/12

### CODE REVIEW

**Review by loukou007 — COMMENTED** [review]
Thank you @ming-tung for introducing this! It's pretty awesome and better than the previous approach. Good job.

File: django_fsm_freeze/models.py [hunk 1]
```diff
@@ -13,6 +14,26 @@ class FreezeValidationError(ValidationError):
...
+    for obj in objs:
+        if not isinstance(obj, FreezableFSMModelMixin):
+            raise FreezeValidationError(
+                f'Unsupported argument {obj!r}. '
+                f'`bypass_fsm_freeze()` accepts instance(s) from '
+                f'FreezableFSMModelMixin.'
+            )
+    for obj in objs:
+        obj._bypass_fsm_freeze = True
+    yield
```
Comment (ticosax) [inline django_fsm_freeze/models.py:32]: we need try finally block

File: django_fsm_freeze/models.py [hunk 2]
```diff
@@ -13,6 +14,26 @@ class FreezeValidationError(ValidationError):
     pass
 
 
+@contextmanager
+def bypass_fsm_freeze(objs: Union[object, list]):
+    if isinstance(objs, list):
+        pass
+    else:
+        objs = [objs]
```
Comment (ticosax) [inline django_fsm_freeze/models.py:22]:
```suggestion
    if not isinstance(objs, Iterable):
        objs = (objs,)
```

File: django_fsm_freeze/models.py [hunk 3]
```diff
@@ -13,6 +14,26 @@ class FreezeValidationError(ValidationError):
     pass
 
 
+@contextmanager
+def bypass_fsm_freeze(objs: Union[object, list]):
```
Comment (ticosax) [inline django_fsm_freeze/models.py:18]:
```suggestion
def bypass_fsm_freeze(objs: Union[object, Iterable]):
```

**Review by ticosax — COMMENTED** [review]
Looks good. Have you thought about using a global state in order to make the context manager parameter less ?

The advantage is that all freezable instances in the stack will be included

```python
def so_something():
    obj = Model.objects.get()
    # ...
    obj.save()  # bypassed also

with bypass_fsm_freeze():
    obj.save()  # bypassed 
    do_something()
```

File: django_fsm_freeze/models.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 from collections import defaultdict
+from collections.abc import Iterable
```
Comment (ticosax) [inline django_fsm_freeze/models.py:2]: it's also available from `typing` module. Maybe it's more convenient ?

**Review by ticosax — APPROVED** [review]

**Comment by ming-tung** [comment]
> Looks good. Have you thought about using a global state in order to make the context manager parameter less ?
> 
> The advantage is that all freezable instances in the stack will be included

What do you mean by a global state?

**Comment by ticosax** [comment]
> > Looks good. Have you thought about using a global state in order to make the context manager parameter less ?
> > The advantage is that all freezable instances in the stack will be included
> 
> What do you mean by a global state?

instead of using the instance to store the sate of the manager, `obj._bypass_fsm_freeze = True`,
We could keep the value of `_bypass_fsm_freeze` as a global variable, thus removing the need to pass the objs to the context manager.
```python
with bypass_fsm_freeze(obj):
```
vs.
```python
with bypass_fsm_freeze():
```

**Comment by ming-tung** [comment]
IMO I would try to avoid using global variables :) and it seems to me that whether or not to bypass fsm freeze can be decided on the per-object level which makes it also more explicit, avoiding implicitly bypassing - assuming that users would most of the time want to have the freeze check, and only on some special condition/reason, use the contextmanager to bypass.

**Comment by ming-tung** [comment]
I will merge this now. We can still revisit this design in the future when there is a need/argument.

**Pull request merged**